### PR TITLE
fix securityConfigSecrets.config.data secrets mount plus permissions

### PR DESCRIPTION
**Pull request by alborotogarcia**
### Description
[Describe what this change achieves]
Check volume mount permissions to 1000
Fix securityConfigSecrets mounts
### Issues Resolved
[List any issues this PR will resolve]
 #8 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: charts/opensearch/templates/securityconfig.yaml [hunk 1]
```diff
@@ -5,9 +5,9 @@ metadata:
   name: {{ .Values.securityConfig.config.securityConfigSecret }}
   namespace: {{ .Release.Namespace }}
   labels:
-    app: {{ .Chart.Name }}
-    release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
+    app: "{{ .Chart.Name }}"
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
```
Comment (DandyDeveloper) [inline charts/opensearch/templates/securityconfig.yaml:10]: Change these to the labels template. Technically `app` has been replaced by a newer label in Helm.

File: charts/opensearch/values.yaml [hunk 2]
```diff
@@ -196,10 +196,17 @@ extraContainers: []
...
+ - name: volume-chmod
+   image: busybox
+   command: ['sh', '-c']
+   args:
+     - 'chown -R 1000:1000 /usr/share/opensearch'
+   securityContext:
+     runAsUser: 0
+   volumeMounts:
+     - mountPath: /usr/share/opensearch/data
+       name: opensearch-cluster-master
```
Comment (DandyDeveloper) [inline charts/opensearch/values.yaml:209]: If this is necessary, then it should be default to the statefulset.

**Review by DandyDeveloper — CHANGES_REQUESTED** [review]

File: charts/opensearch/templates/securityconfig.yaml (hunk 1 above)
Comment (alborotogarcia) [reply charts/opensearch/templates/securityconfig.yaml]: Yes you're right! I didn't check twice!

File: charts/opensearch/values.yaml (hunk 2 above)
Comment (alborotogarcia) [reply charts/opensearch/values.yaml]: done !

**Comment by peterzhuamazon** [comment]
Need some reviews @DandyDeveloper @TheAlgo @smlx very much thanks :)

File: charts/opensearch/values.yaml [hunk 3]
```diff
@@ -196,11 +196,6 @@ extraContainers: []
   #   image: busybox
   #   command: ['do', 'something']
 
-extraInitContainers: []
```
Comment (DandyDeveloper) [inline charts/opensearch/values.yaml:199]: We should keep this @alborotogarcia

**Review by DandyDeveloper — CHANGES_REQUESTED** [review]
We need the `extraInitContainers` but empty:

```
extraInitContainers: []
```

File: charts/opensearch/values.yaml (hunk 3 above)
Comment (alborotogarcia) [reply charts/opensearch/values.yaml]: I forgot to add it back nice catch @DandyDeveloper !

**Comment by peterzhuamazon** [comment]
Hi @DandyDeveloper @TheAlgo Please review again seems there are changes from @alborotogarcia.
Thanks.

File: charts/opensearch/templates/statefulset.yaml [hunk 4]
```diff
@@ -211,8 +211,20 @@ spec:
       {{- if .Values.hostAliases }}
       hostAliases: {{ toYaml .Values.hostAliases | nindent 8 }}
       {{- end }}
-      {{- if or (.Values.extraInitContainers) (.Values.keystore) }}
       initContainers:
+{{- if .Values.persistence.enabled }}
+      - name: fsgroup-volume
+        image: busybox
+        command: ['sh', '-c']
```
Comment (DandyDeveloper) [inline charts/opensearch/templates/statefulset.yaml:218]: Won't this break? `initContainers` will be written but if `persistence.enabled=`false`, `extraInitContainers=[]` AND `keystore.enabled=false` are missing. It's empty in the YAML.

**Review by DandyDeveloper — CHANGES_REQUESTED** [review]
@alborotogarcia Sorry to be so pedantic here.

File: charts/opensearch/templates/statefulset.yaml (hunk 4 above)
Comment (alborotogarcia) [reply charts/opensearch/templates/statefulset.yaml:219]: indeed @DandyDeveloper !  it should be init if any of them applies, now should be fixed

**Review by DandyDeveloper — APPROVED** [review]

**Review by TheAlgo — APPROVED** [review]
LGTM!!

**Comment by sastorsl** [comment]
Regarding https://github.com/opensearch-project/helm-charts/pull/9/files#diff-08885706cd45ad445d2696cd7e39cf7d4b5e82cb51b0ee6bd86e6ee9f818e158R218

It would be nice if it was possible to override the image from values. For instance if one has a local image proxy with proper credentials setup.

**Pull request merged**